framework/src/base/FEProblemBase.C

```diff
@@ -364,6 +364,8 @@ void FEProblemBase::initialSetup()
   if (_app.isRecovering() && (_app.isUltimateMaster() || _force_restart))
     _resurrector->setRestartFile(_app.getRecoverFileBase());
 
+  _mesh.initialSetup();
```

Comment:
The mesh data gets recovered during `MooseMesh::init()` during a recovery, which happens long before `FEProblem::initialSetup()` gets called.

In our case, we need to reconstruct some auxiliary data structures that are specific to peridynamics that live in the peridynamics mesh object.